### PR TITLE
Dp/adjust position proxy action

### PR DESCRIPTION
**Pull request by paszkowskiDamian**
# [Integrate adjust position proxy actions](https://app.clubhouse.io/oazo-apps/story/2183/connect-adjust-position-actions-to-proxy-actions-smart-contract)
  
## Changes 👷‍♀️
 - make a call to blockchain to adjust position - multiply/demultiply 
  
## How to test 🧪
- setup hardhat locally
- deploy proxy actions
- use form to adjust position

Aug-10-2021 14-32-35

    
## Definition of done ✔️

- [ ] Acceptance criteria for each issue met
- [ ] Unit tests written where needed and passing
- [ ] End-to-end tests for happy path
- [ ] Documentation updated <When applicable>
- [ ] Project builds without errors
- [ ] Non-functional requirements met
- [ ] Code reviewed and functionality tested by the reviewer (locally, Heroku, etc.)
- [ ] Feature verified and accepted by Product Owner
- [ ] Project deployed to production environment

### CODE REVIEW

File: features/manageMultiplyVault/manageMultiplyVaultTransitions.ts [hunk 1]
```diff
@@ -184,18 +184,13 @@ export function applyManageVaultTransition(
...
-  // const { depositAmount, generateAmount } = state
-  // const isDepositAndGenerate = depositAmount || generateAmount
-
-  return manageVaultDepositAndGenerate(txHelpers$, change, state)
-  // if (true) {
-  //   return manageVaultDepositAndGenerate(txHelpers$, change, state)
-  // } else {
-  //   return manageVaultWithdrawAndPayback(txHelpers$, change, state)
-  // }
+  // state.exchangeAction === 'BUY_COLLATERAL'
```
Comment (johnnieskywalker) [inline features/manageMultiplyVault/manageMultiplyVaultTransitions.ts:192]: [minor] please remove comment if it won't be needed soon

**Review by johnnieskywalker — APPROVED** [review]

File: features/manageMultiplyVault/manageMultiplyVaultTransactions.ts [hunk 2]
```diff
@@ -227,23 +227,33 @@ export function applyManageVaultTransaction(
...
       first(),
       switchMap(({ sendWithGasEstimation }) =>
         sendWithGasEstimation(adjustMultiplyVault, {
           kind: TxMetaKind.adjustPosition,
-          depositCollateral: zero,
-          requiredDebt: zero,
-          borrowedCollateral: zero,
-          userAddress: '0x',
+          depositCollateral: depositAmount || zero,
+          requiredDebt: debtDelta?.abs() || zero,
```
Comment (adamskrodzki) [inline features/manageMultiplyVault/manageMultiplyVaultTransactions.ts:249]: shouldn't be current debt added somewhere here ? or just naming is wrong?

File: features/manageMultiplyVault/manageMultiplyVaultTransactions.ts [hunk 3]
```diff
@@ -227,23 +227,33 @@ export function applyManageVaultTransaction(
...
+          borrowedCollateral: collateralDelta?.abs() || zero,
```
Comment (adamskrodzki) [inline features/manageMultiplyVault/manageMultiplyVaultTransactions.ts:250]: same as above

**Review by adamskrodzki — CHANGES_REQUESTED** [review]
I think if not changes then at least some explanations

**Comment by adamskrodzki** [comment]
Doesn't this stuff sopposed to have some tests ?

**Review by pacxiu — DISMISSED** [review]

File: features/manageMultiplyVault/manageMultiplyVaultTransactions.ts (hunk 2 above)
Comment (paszkowskiDamian) [reply features/manageMultiplyVault/manageMultiplyVaultTransactions.ts:249]: I thought the contract accepts the debt change and collateral change, I can change to calculated `afterDebt` value

Comment (adamskrodzki) [reply features/manageMultiplyVault/manageMultiplyVaultTransactions.ts:249]: ok, naming is wrong on SC side

File: features/manageMultiplyVault/manageMultiplyVaultTransactions.ts (hunk 3 above)
Comment (adamskrodzki) [reply features/manageMultiplyVault/manageMultiplyVaultTransactions.ts:250]: again naming is wrong on SC side

**Comment by paszkowskiDamian** [comment]
From this point you will need to set 

```
MULTIPLY_PROXY_ACTIONS=
EXCHANGE=
```

addresses in `.env`

File: features/manageMultiplyVault/manageMultiplyVaultTransitions.ts (hunk 1 above)
Comment (paszkowskiDamian) [reply features/manageMultiplyVault/manageMultiplyVaultTransitions.ts:192]: It will be removed in next iteration ;)

**Pull request merged**